tests/Operation/FindAndModifyTest.php

```diff
@@ -83,4 +84,46 @@ public function testConstructorUpdateAndRemoveOptionsAreMutuallyExclusive(): voi
         $this->expectExceptionMessage('The "remove" option must be true or an "update" document must be specified, but not both');
         new FindAndModify($this->getDatabaseName(), $this->getCollectionName(), ['remove' => true, 'update' => []]);
     }
+
+    public function testExplainableCommandDocument(): void
+    {
+        // all options except deprecated "snapshot" and "maxScan"
+        $options = [
+            'arrayFilters' => [['x' => 1]],
+            'bypassDocumentValidation' => true,
+            'collation' => ['locale' => 'fr'],
+            'comment' => 'explain me',
+            'fields' => ['_id' => 0],
+            'hint' => '_id_',
+            'maxTimeMS' => 100,
+            'new' => true,
+            'query' => ['y' => 2],
+            'remove' => false,
+            'sort' => ['x' => 1],
+            'typeMap' => ['root' => 'array'],
+            'update' => ['$set' => ['x' => 2]],
+            'upsert' => true,
+            'let' => ['a' => 3],
+            'writeConcern' => new WriteConcern(WriteConcern::MAJORITY),
+        ];
+        $operation = new FindAndModify($this->getDatabaseName(), $this->getCollectionName(), $options);
+
+        $expected = [
+            'findAndModify' => $this->getCollectionName(),
+            'new' => true,
+            'upsert' => true,
+            'collation' => (object) ['locale' => 'fr'],
+            'fields' => (object) ['_id' => 0],
+            'let' => (object) ['a' => 3],
```

Comment:
`let` option is provided here, but skipped for `delete` operation. I'm not sure if it's expected or a bug?

`FindAndModify::getCommandDocument()` returns unchanged command, it's modified for other operations.

Reply:
`let` should not be skipped for delete operations. The best course of action would be to add specific tests for the various `FindOneAnd...` operation classes and assert that they behave correctly.

That said, I looked through both `FindAndModify` and `FindOneAndDelete` operation classes and the `let` option seems to be handled correctly. What do you mean with "it's modified for other operations"?

Reply:
`FindAndModify::getCommandDocument` is a direct copy of `createCommandDocument`.

https://github.com/mongodb/mongo-php-library/blob/b3607d91b71b4efb2d92e52e68073308c9696b39/src/Operation/FindAndModify.php#L263-L266

In other Operation classes, the `getCommandDocument` is customized for explain needs.
https://github.com/mongodb/mongo-php-library/blob/b3607d91b71b4efb2d92e52e68073308c9696b39/src/Operation/Delete.php#L178-L181

https://github.com/mongodb/mongo-php-library/blob/b3607d91b71b4efb2d92e52e68073308c9696b39/src/Operation/Find.php#L331-L341

Reply:
> FindAndModify::getCommandDocument is a direct copy of createCommandDocument.

This is generally the case for most operations that are executed using `executeCommand()` (or a related function). An exception is where some `execute` options need to be injected, as you recently did with adding `readConcern` and removing `writeConcern`.

Operations using `executeBulkWrite()` and `executeQuery()` are special because PHPLIB has no reason to construct the outgoing command apart from `Explainable::getCommandDocument()`.

In the case of Find, it's a bit odd that `createCommandDocument()` only exists for the purpose of `getCommandDocument()`. It looks like that dates back to [c7b2b03](https://github.com/mongodb/mongo-php-library/commit/c7b2b033b327b29a619ee0ed2bcd9d055b4ab9d1#diff-ff201f3e96f2f4300f6d25d87430d60784224c2b5509ed1f4d52d5ab03d52d92R304). You're welcome to factor out that private method if you wish (either in this PR or down the line).